### ISSUE TITLE
Do you want ECM?

### ISSUE THREAD

**Issue by darloth** [Alarcarr/Rising-Star#21]
This came out of my renewed investigation into the plausibility of Point-Defense type weaponry in Heralds.  

It's not really very doable in a standalone, modular mod, , but given that Rising-Star is essentially a total conversion that can touch every weapon file and combat script if it needs to, two types appear to be plausible now:

- **Passive ECM** - this would be a system you add to a ship which would give a percentile chance to evade all damage from any weapon tagged with, say, SeekingMunition (or possibly using different effectors).  This would be done by using one of the currently unused DamageFlags so that the receiving ship can tell it's just been hit by something that is potentially jammable.  The ECM system would then have a high resolution priority (above shields even) and would resolve first, giving a percentage chance to block based on some value or another.  With some finagling, you could maaaybe have ECCM or at least some things being harder to block than others, but even if it were simply a flat percentage it would be workable.

- **Active ECM** - This would simply be a weapon type that targets only weapons tagged with SeekingMunition, and targets them through all forms of defense.  It would degrade the effectiveness of the entire system all at once, which in practice reduces the launcher's RoF, representing the jamming effect quite nicely and also (unlike Passive ECM) giving an obvious visual change.  The only potential downside is it would cost the enemy supplies as they attempt to repair their systems, unless significant coding were undertaken to fix this.  I don't personally think it's a downside, especially if, for example, the ECM uses a special effector that can't reduce the target system HP below 50% or whatever.

**Comment by DaloLorn** [comment]
I thought I posted here. (Also, it's Rising Stars, with an s and without a dash.)

Anyway, we don't really have a lot of damage flags to spare anymore. IIRC there's only 8 of them, we use seven or so.

**Comment by darloth** [comment]
Fair enough if there's none left, I just still feel missiles don't feel significantly different as weapon types... but they do have just barely enough to differentiate them given they don't miss.   Also, the repo is Rising-Star.so I used that.  I'll try and remember that Rising Stars is the official name in future :)

**Comment by Alarcarr** [comment]
I think its a good thing to have ,but we need to clean all the trash we already have.Balancing all in a single wave will be deadly.
